### PR TITLE
added quick globbing support for includes, does not satisfy vimeo/psalm#84 but is sort of related

### PR DESCRIPTION
**Pull request by SignpostMarv**
Because I wanted to do this:
```xml
<?xml version="1.0"?>
<psalm
    name="Example Psalm config with recommended defaults"
    stopOnFirstError="false"
    useDocblockTypes="true"
    totallyTyped="false"
>
    <projectFiles>
        <directory name="./vendor/foo/*/src" />
    </projectFiles>
```

### CODE REVIEW

File: src/Psalm/Config/FileFilter.php [hunk 1]
```diff
@@ -59,6 +59,23 @@ public static function loadFromXMLElement(
         if ($e->directory) {
             /** @var \SimpleXMLElement $directory */
             foreach ($e->directory as $directory) {
+                if (strpos((string)$directory['name'], '*') !== false) {
```
Comment (muglug) [inline src/Psalm/Config/FileFilter.php]: How about
```php
$prospective_directory_path = getcwd() . DIRECTORY_SEPARATOR . $directory['name'];
if (strpos($prospective_directory_path, '*') !== false) {
```
and then you can use `$prospective_directory_path` below

File: src/Psalm/Config/FileFilter.php [hunk 2]
```diff
@@ -59,6 +59,23 @@ public static function loadFromXMLElement(
...
+                    $globs = array_map(
+                        'realpath',
+                        array_filter(
+                            glob(getcwd() . DIRECTORY_SEPARATOR . (string)($directory['name'])),
+                            'is_dir'
+                        )
+                    );
+                    foreach ($globs as $globIndex => $directory_path) {
```
Comment (muglug) [inline src/Psalm/Config/FileFilter.php]: could you use `$snake_case` here please?

Comment (SignpostMarv) [inline src/Psalm/Config/FileFilter.php]: addressed in 4c181f2

File: src/Psalm/Config/FileFilter.php (hunk 1 above)
Comment (SignpostMarv) [inline src/Psalm/Config/FileFilter.php]: addressed in 1d6f654

**Pull request merged**